pytest_asyncio/plugin.py

```diff
@@ -517,20 +528,20 @@ def pytest_pycollect_makeitem_preprocess_async_fixtures(
 @pytest.hookimpl(specname="pytest_pycollect_makeitem", hookwrapper=True)
 def pytest_pycollect_makeitem_convert_async_functions_to_subclass(
     collector: Union[pytest.Module, pytest.Class], name: str, obj: object
-) -> Union[
-    pytest.Item, pytest.Collector, List[Union[pytest.Item, pytest.Collector]], None
-]:
+) -> Generator[None, Any, None]:
```

Comment:
The `Any` here can be properly typed with newer pytest and/or pluggy but pytest 7.0 is too old for it, so I made it `Any` for now.

Reply:
Thanks for the hint. I opened an issue for dropping pytest 7 support in the future and added a corresponding comment in the code.